File: python3_solution_set2/2393. Count Strictly Increasing Subarrays.py
class Solution:
    def countSubarrays(self, nums: List[int]) -> int:
        ## simple dp
        n = len(nums)
        dp = [1 for x in range(n)]
        
        for x in range(1,n):
            if nums[x] > nums[x-1]:
                dp[x] = 1 + dp[x-1]
            else:
                dp[x] = 1
        return sum(dp)
                
        

File: cpp_solutions/2393. Count Strictly Increasing Subarrays.cc
class Solution {
public:
    long long countSubarrays(vector<int>& nums) {
        int n = nums.size();
        vector<int> dp(n,1);
        for(int x =1 ; x < n ; ++x) {
            if (nums[x] > nums[x-1]) {
                dp[x] = 1 + dp[x-1];
            } 
        }
        long long ans = accumulate(dp.begin(), dp.end(), 0LL);
        return ans;
    }
};
